packages/core-compat-api/src/collectLegacyRoutes.test.tsx

```diff
@@ -279,4 +279,58 @@ describe('collectLegacyRoutes', () => {
       screen.findByText('plugins: test'),
     ).resolves.toBeInTheDocument();
   });
+
+  it('should throw if invalid Route has been detected', async () => {
+    const plugin = createPlugin({
+      id: 'test',
+    });
+    const routeRef = createRouteRef({ id: 'test' });
+    const Page = plugin.provide(
+      createRoutableExtension({
+        name: 'Test',
+        mountPoint: routeRef,
+        component: () =>
+          Promise.resolve(() => {
```

Comment:
very matter of taste, feel free to ignore:

```suggestion
        component: async () => () => {
```

haha i know you're just being consistent with the others, but there's something about Promise.resolve that just bugs me :) same below